src/main/kotlin/com/amazon/opendistroforelasticsearch/indexstatemanagement/models/actions/ActionRetry.kt

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazon.opendistroforelasticsearch.indexstatemanagement.models.actions
+
+import org.elasticsearch.common.unit.TimeValue
+import org.elasticsearch.common.xcontent.ToXContent
+import org.elasticsearch.common.xcontent.ToXContentObject
+import org.elasticsearch.common.xcontent.XContentBuilder
+import org.elasticsearch.common.xcontent.XContentParser
+import org.elasticsearch.common.xcontent.XContentParser.Token
+import org.elasticsearch.common.xcontent.XContentParserUtils.ensureExpectedToken
+import java.io.IOException
+
+data class ActionRetry(
+    val count: Long? = null,
+    val backoff: String = DEFAULT_BACKOFF,
+    val delay: String? = null
+) : ToXContentObject {
+
+    init {
+        if (count != null) require(count > 0) { "Count for ActionRetry must be greater than 0" }
+
+        if (delay != null) {
+            try {
+                TimeValue.parseTimeValue(delay, "")
+            } catch (e: IllegalArgumentException) {
+                throw IllegalArgumentException("Must have a valid delay for ActionRetry")
+            }
+        }
+    }
+
+    override fun toXContent(builder: XContentBuilder, params: ToXContent.Params): XContentBuilder {
+        builder
+            .startObject()
+                .field(COUNT_FIELD, count)
+                .field(BACKOFF_FIELD, backoff)
+                .field(DELAY_FIELD, delay)
```

Comment:
If the fields can be null, we should check if they are null before writing in the XContent.

Reply:
Changed ```ActionRetry``` so that all three fields are now non-nullable so this change is no longer necessary.